File: JUnitTestingLastAct/src/Decorator/RedShapeDecorator.java
package Decorator;

// Concrete class extending the abstract class
public class RedShapeDecorator extends ShapeDecorator{

    public RedShapeDecorator(Shape decoratedShape) {
        super(decoratedShape);
    }

    @Override
    public void draw() {
        decoratedShape.draw();
        setRedBorder(decoratedShape);
    }

    // Display message whenever function is called
    private void setRedBorder (Shape decoratedShape){
        System.out.println("Border Color: Red");
    }
}

File: KotlinTesting/src/Design Patterns/Structural/Decorator/RedShapeDecorator.kt
package Decorator

// Concrete class extending the abstract class
open class RedShapeDecorator(decoratedShape: Shape?) : ShapesDecorator(decoratedShape) {
    override fun draw() {
        decoratedShape?.draw()
        decoratedShape?.let { setRedBorder(it) }
    }

    // Display message whenever function is called
    private fun setRedBorder(decoratedShape: Shape) {
        println("Border Color: Red")
    }
}